djed-sdk/dist/esm/index.js

```diff
@@ -1067,35 +1068,126 @@ var coinArtifact = {
+      suffix = suffix.replace(/,/g, "");
+    }
+  
+    prefix = reverseString(intersperseCommas(reverseString(prefix)));
+  
+    return prefix + "." + suffix;
+  }
+  
+  function scaledUnscaledPromise(promise, scaling) {
+    return promise.then((value) => [decimalScaling(value.toString(10), scaling), value]);
+  }
 
-// Modify getDjedContract to accept DJED_ADDRESS as a parameter
-const getDjedContract = (web3, DJED_ADDRESS) => {
-  return new web3.eth.Contract(djedArtifact.abi, DJED_ADDRESS);
-};
+const DjedInstance = async (BLOCKCHAIN_URI, DJED_ADDRESS) => {
+  try {
+    
+    const web3 = new Web3(new Web3.providers.HttpProvider(BLOCKCHAIN_URI));
 
-const getOracleAddress = async (djedContract) => {
-  return await djedContract.methods.oracle().call();
+    
+    const djedContract = new web3.eth.Contract(djedArtifact.abi, DJED_ADDRESS);
+
+    
+    const [oracleAddress, stableCoinAddress, reserveCoinAddress] = await Promise.all([
+      djedContract.methods.oracle().call(), // Fetch the Oracle address
+      djedContract.methods.stableCoin().call(), // Fetch the StableCoin address
+      djedContract.methods.reserveCoin().call() // Fetch the ReserveCoin address
+    ]);
+
+   
+    const stableCoin = new web3.eth.Contract(coinArtifact.abi, stableCoinAddress);
+    const reserveCoin = new web3.eth.Contract(coinArtifact.abi, reserveCoinAddress);
+
+    
+    const [scDecimals, rcDecimals] = await Promise.all([
+      convertInt(web3Promise(stableCoin, "decimals")),
+      convertInt(web3Promise(reserveCoin, "decimals"))
+    ]);
+
+    
+    return {
+      web3, 
+      djedContract, 
+      oracleAddress, 
+      stableCoin, 
+      reserveCoin, 
+      scDecimals, 
+      rcDecimals 
+    };
+  } catch (error) {
+    // If any error occurs during the process, throw an error with a descriptive message.
+    throw new Error(`Initialization failed: ${error.message}`);
+  }
 };
 
-const getCoinContracts = async (djedContract, web3) => {
-  const [stableCoinAddress, reserveCoinAddress] = await Promise.all([
-    djedContract.methods.stableCoin().call(),
-    djedContract.methods.reserveCoin().call()
+const BC_DECIMALS = 18;
+
+const getAccountDetails = async (account, djedInstance) => {
+  const { web3, stableCoin, reserveCoin, scDecimals, rcDecimals } = djedInstance;
+
+  const [
+    [scaledBalanceSc, unscaledBalanceSc],
+    [scaledBalanceRc, unscaledBalanceRc],
+    [scaledBalanceBc, unscaledBalanceBc]
+  ] = await Promise.all([
+    scaledUnscaledPromise(web3Promise(stableCoin, "balanceOf", account), scDecimals),
+    scaledUnscaledPromise(web3Promise(reserveCoin, "balanceOf", account), rcDecimals),
+    scaledUnscaledPromise(web3.eth.getBalance(account), BC_DECIMALS)
   ]);
-  const stableCoin = new web3.eth.Contract(coinArtifact.abi, stableCoinAddress);
-  const reserveCoin = new web3.eth.Contract(coinArtifact.abi, reserveCoinAddress);
-  return { stableCoin, reserveCoin };
+
+  return {
+    scaledBalanceSc,
```

Comment:
It seems that "unscaled..." is a raw number and "scaled..." is the number converted to a string, according to the number of decimals. Doing this conversion at the very beginning and passing around "scaled" and "unscaled" versions of the same data is confusing, redundant and error prone. There must be a more elegant way to do this.

@Tanya-ruby , improving this is beyond the scope of this PR. But could you keep this in mind and try to improve this when you have an opportunity? I suggest creating an issue for this and one day we can discuss how to solve it.

Reply:
@ceilican yes ! Completely makes sense ,it can also create a lot of confusion while using the SDK.
With respect to parameters and the values to be sent ! 
I will workaround the issue and find a better way this can be dealt with.